cmd/psa/verify_as_relyingparty.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package psa
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/veraison/apiclient/verification"
+	"github.com/veraison/evcli/common"
+	"github.com/veraison/psatoken"
+)
+
+var (
+	relyingPartyTokenFile *string
+	relyingPartyAPIURL    *string
+)
+
+var (
+	relyingPartyVeraisonClient common.IVeraisonClient = &verification.ChallengeResponseConfig{}
+	relyingPartyCmd                                   = NewRelyingPartyCmd(common.Fs, relyingPartyVeraisonClient)
+)
+
+func NewRelyingPartyCmd(fs afero.Fs, veraisonClient common.IVeraisonClient) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "relying-party",
+		Short: "Emulate a Relying Party",
+		Long: `
+	This command implements the "relying party mode" of interaction, where the
+	relying party was the challenger, and therefore the nonce is provided by the
+	caller implicitly in an already well-formed and signed PSA attestation
+	token, possibly produced by a previous invocation to "evcli psa create".
+	
+		evcli psa verify-as relying-party \
+		              --api-server=https://veraison.example/challenge-response/v1 \
+		              --token=psa-token.cbor
+	
+	`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			token, err := afero.ReadFile(fs, *relyingPartyTokenFile)
+			if err != nil {
+				return err
+			}
+
+			var e psatoken.Evidence
+
+			if err = e.FromCOSE(token, psatoken.PSA_PROFILE_1, psatoken.PSA_PROFILE_2); err != nil {
+				return err
+			}
+
+			nonce, err := e.Claims.GetNonce()
+			if err != nil {
+				return err
+			}
+
+			if err = veraisonClient.SetNonce(nonce); err != nil {
+				return err
+			}
```

Comment:
Should we check on Line 63, valid URI before passing?

Reply:
apiclient does that, so here we are OK to swallow anything that is a string